internal/pkg/net/blocksub/topic.go

```diff
@@ -2,17 +2,48 @@ package blocksub
 
 import (
 	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/filecoin-project/go-filecoin/internal/pkg/block"
+	e "github.com/filecoin-project/go-filecoin/internal/pkg/enccid"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/encoding"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/types"
 )
 
 // BlockTopic returns the network pubsub topic identifier on which new blocks are announced.
 func Topic(networkName string) string {
 	return fmt.Sprintf("/fil/blocks/%s", networkName)
 }
 
-// Coming soon:
-//type Message struct {
-//	_           struct{} `cbor:",toarray"`
-//	Header      block.Block
-//	BLSMsgCids  []e.Cid
-//	SECPMsgCids []e.Cid
-//}
+type Payload struct {
+	_           struct{} `cbor:",toarray"`
+	Header      block.Block
+	BLSMsgCids  []e.Cid
+	SECPMsgCids []e.Cid
+}
+
+func MakePayload(header *block.Block, BLSMessages, SECPMessages []*types.SignedMessage) ([]byte, error) {
```

Comment:
It would be slightly better to pass BLSMessages as unsigned and SECPMessages as signed since they are already separated and the the bls message signatures are unused

Reply:
You could also argue that only the two cid slices should be output from the mining worker

Reply:
On the latter point I consider the abstraction cleaner by returning the messages, and it doesn't cost anything extra. The question is where should the knowledge about how to extract pubsub-relevant info from the messages go? I don't think the block producer should know/care.